### DISCUSSION TITLE
Should BaseResponse Error.Code be string or int? Compatibility with LiteLLM

### DISCUSSION THREAD

**Discussion by repko-artem** [betalgo/openai#615] [Q&A] [answered]
Hi.
I have troubles when using LiteLLM proxy as OpenAI API.
The problem is that Betalgo/OpenAI expects Error.Code to be only a string, but LiteLLM returns that code as integer:
```
{
	"error": {
		"message": "Authentication Error, No connected db.",
		"type": "auth_error",
		"param": "None",
		"code": 401
	}
}
```

I created question for LiteLLM:
https://github.com/BerriAI/litellm/discussions/4834#discussioncomment-10192506

But also I want to ask you, what is the right type for this code: string or int? Is there any official API for it? Maybe you can support both variants?

**Comment by kayhantolga** [comment, marked as answer]
Hi, OpenAI error codes are strings, which is why the SDK expects a string error code. Here is a sample error response from the OpenAI API.

```json
{
  "error": {
    "message": "The model `sdsad` does not exist or you do not have access to it.",
    "type": "invalid_request_error",
    "param": null,
    "code": "model_not_found"
  }
}
```

**Comment by repko-artem** [comment]
Hi, @kayhantolga 
Thanks! Also I looked on OpenAI python official client and it's declared as string there too:
https://github.com/openai/openai-python/blob/195c05a64d39c87b2dfdf1eca2d339597f1fce03/src/openai/types/shared/error_object.py#L11
